### PR TITLE
Provide pageable users of space endpoint

### PR DESCRIPTION
**Pull request by moleske**
- /v3/spaces/:guid/users now exists with filters

[#2489]

Thanks for contributing to cloud_controller_ng. To speed up the process of reviewing your pull request please provide us with:

* A short explanation of the proposed change:

* An explanation of the use cases your change solves

* Links to any other associated PRs

* [x] I have reviewed the [contributing guide](https://github.com/cloudfoundry/cloud_controller_ng/blob/main/CONTRIBUTING.md)

* [x] I have viewed, signed, and submitted the Contributor License Agreement

* [x] I have made this pull request to the `main` branch

* [x] I have run all the unit tests using `bundle exec rake`

* [ ] I have run [CF Acceptance Tests](https://github.com/cloudfoundry/cloud_controller_ng#cf-acceptance-tests-cats)

@ctlong

### CODE REVIEW

File: spec/request/spaces_spec.rb [hunk 1]
```diff
@@ -1196,6 +1196,301 @@
...
+          VCAP::CloudController::User.plugin :timestamps, update_on_create: true
+        end
+
+        it 'returns 200 and filters' do
+          org.add_user(resource_1)
+          space1.add_manager(resource_1)
+          get "/v3/spaces/#{space1.guid}/users?updated_ats[lt]=#{resource_3.updated_at.iso8601}", nil, admin_headers
+
+          expect(last_response).to have_status_code(200)
+          # TODO(carson and michael): do we include user.guid and client.guid or not?
```
Comment (sweinstein22) [inline spec/request/spaces_spec.rb:1368]: 👀

File: app/fetchers/space_users_list_fetcher.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            label_klass: UserLabelModel,
+            resource_dataset: dataset,
+            requirements: message.requirements,
+            resource_klass: User,
+          )
+        end
+
+        dataset = super(message, dataset, User)
+
+        dataset.inner_join(SpaceDeveloper.table_name, user_id: :id).where(space_id: space.id).select_all(:users).
```
Comment (sweinstein22) [inline app/fetchers/space_users_list_fetcher.rb:29]: Does it make more sense to keep this logic at the model layer? We already have an [established pattern for visible org users](https://github.com/cloudfoundry/cloud_controller_ng/blob/2860386d3f0a8402df451a52858df321f104fd44/app/models/runtime/user.rb#L188)

**Review by sweinstein22 — COMMENTED** [review]
### Behavior Check

- deployed changes to a bosh lite
- ran `cf curl /v3/users` and `cf curl /v3/spaces/:guid/users` to confirm that only a subset of users was returned. User missing from the space level call was the `network-policy`

```
$ cf curl /v3/spaces/369cc547-b6b8-4ccb-a3c3-987845b04049/users | grep total_results
    "total_results": 3,

$ cf curl /v3/users | grep total_results
    "total_results": 4,
```
- confirmed that permissions for each role are correct, results are identical for all roles except OrgAuditor, which receives a `404`. This matches the error code received by OrgAuditors when curling `/v3/spaces/:guid`.

File: app/fetchers/space_users_list_fetcher.rb (hunk 2 above)
Comment (moleske) [reply app/fetchers/space_users_list_fetcher.rb:29]: I didn't look at user.rb before.  Cool with me to move it if that's the pattern we like

**Review by sweinstein22 — APPROVED** [review]
Running through acceptance again as the implementation changed pretty significantly.

### Behavior Check

As a space supporter, confirmed that the list of users returned for a space is a subset of those returned from `/v3/users`:
```
$ cf curl /v3/spaces/369cc547-b6b8-4ccb-a3c3-987845b04049/users | grep total_results
    "total_results": 6,
$ cf curl /v3/users | grep total_results
    "total_results": 8,
```
Also confirmed that output of endpoint looked correct, and then checked to make sure there wasn't a diff in output between roles.
```
$ diff space-supporter space-manager

$ diff space-supporter space-dev

$ diff space-supporter space-auditor

$ diff space-supporter org-manager

```
Confirmed that org auditors get a `404`:
```
$ cf curl /v3/spaces/369cc547-b6b8-4ccb-a3c3-987845b04049/users -v
REQUEST: [2021-09-08T09:58:20-07:00]
GET /v3/spaces/369cc547-b6b8-4ccb-a3c3-987845b04049/users HTTP/1.1
...

RESPONSE: [2021-09-08T09:58:20-07:00]
HTTP/1.1 404 Not Found
...
{
  "errors": [
    {
      "code": 10010,
      "detail": "Space not found",
      "title": "CF-ResourceNotFound"
    }
  ]
}
```

### Code Check

Looks good! Thanks for rearranging things so we would have more consistent patterns, and for introducing a new model to minimize how much data processing we're doing

### Docs Check

New endpoint is documented, renders as expected

**Pull request merged**